tests/unit/ad/test__ad_utils.py

```diff
@@ -0,0 +1,95 @@
+"""Test suite for ad utils.
+
+The main checks performed are:
+    test_get_set_solution_values: Test of functionality related to setting and getting
+        values from the data dictionary.
+
+"""
+
+import numpy as np
+import pytest
+
+import porepy as pp
+
+
+def test_get_set_solution_values():
+    data = {}
+
+    values_0 = np.linspace(0, 10, 11)
+    values_1 = 2 * np.linspace(0, 10, 11)
+    values_2 = 3 * np.linspace(0, 10, 11)
+    values_3 = 4 * np.linspace(0, 10, 11)
+
+    parameter_name = "parameter_name"
+
+    pp.set_solution_values(
+        name=parameter_name, values=values_0, data=data, time_step_index=0
+    )
+    pp.set_solution_values(
+        name=parameter_name, values=values_1, data=data, iterate_index=0
+    )
+    pp.set_solution_values(
+        name=parameter_name, values=values_2, data=data, time_step_index=1
+    )
+    pp.set_solution_values(
+        name=parameter_name, values=values_3, data=data, iterate_index=1
+    )
+
+    # Values gathered from data dictionary directly
+    values_0_gathered_directly = data[pp.TIME_STEP_SOLUTIONS][parameter_name][0]
```

Comment:
Can't you just use values_0 etc. for comparison below? Is there any reason to retrieve them from the dict?

Reply:
My line of thought was to test the set and get methods somewhat separately. Retrieving values directly from the dictionary shows that the set-method works without relying on the get method to work. But I see now that the get-method test somewhat relies on that the set method works.. Maybe I should just skip the intermediate step and get rid of the directly gathered values.

Reply:
Or you keep them and test that values_0_gathered_directly == values_0

Reply:
Right! Sorry, I missed it. In that case, let's keep it as is.